llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -6890,11 +6890,9 @@ bool AMDGPULegalizerInfo::legalizeWaveID(MachineInstr &MI,
     return false;
   LLT S32 = LLT::scalar(32);
   Register DstReg = MI.getOperand(0).getReg();
-  Register TTMP8 =
-      getFunctionLiveInPhysReg(B.getMF(), B.getTII(), AMDGPU::TTMP8,
-                               AMDGPU::SReg_32RegClass, B.getDebugLoc(), S32);
   auto LSB = B.buildConstant(S32, 25);
   auto Width = B.buildConstant(S32, 5);
+  auto TTMP8 = B.buildCopy(S32, Register(AMDGPU::TTMP8));
```

Comment:
This will build the copy here instead of in the entry block, and will create a new one for each use of the intrinsic. Is that OK?